### PR TITLE
[PW-2390] Add action component on success page if presentToShopper an…

### PR DESCRIPTION
**Pull request by rikterbeek**
…d Action result is provided.

* Added a adyenCheckout3101 requireJS variable to use latest component version. This we can use for all new payment method integrations or upgrades.
* Success block loads all values needed for the component
* Success page renders the component
* Layout renders the styles.css file for showing the component with the right styling

**Description**

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: view/frontend/templates/checkout/success.phtml [hunk 1]
```diff
@@ -25,7 +25,32 @@
...
+        });
+    </script>
+
+
+    <div id="ActionContainer"></div>
+
+
+<?php else: ?>
+
+    <?php if ($block->isBoletoPayment()): ?>
```
Comment (acampos1916) [inline view/frontend/templates/checkout/success.phtml:53]: For phpcs
```suggestion
    </script>
    <div id="ActionContainer"></div>
<?php else: ?>
    <?php if ($block->isBoletoPayment()): ?>
```

Comment (rikterbeek) [reply view/frontend/templates/checkout/success.phtml:53]: yes thanks will fix it

File: view/frontend/templates/checkout/success.phtml [hunk 2]
```diff
@@ -25,7 +25,32 @@
  * @var \Adyen\Payment\Block\Checkout\Success $block
  */
 ?>
-<?php if ($block->isBoletoPayment()): ?>
+
+
+<?php if ($block->renderAction()): ?>
+    <script type="text/javascript">
```
Comment (cyattilakiss) [inline view/frontend/templates/checkout/success.phtml:31]: is there a way to outsource it into a separate JS file?

File: view/frontend/templates/checkout/success.phtml [hunk 3]
```diff
@@ -25,7 +25,32 @@
...
+        require([
+            'jquery',
+            'adyenCheckout3101'  // the alias for "mage/accordion"
+        ], function ($, AdyenCheckout) {
+            var action = JSON.parse('<?php echo $block->getAction(); ?>');
```
Comment (cyattilakiss) [inline view/frontend/templates/checkout/success.phtml:36]: Magento automation check requires this format:

```suggestion
            var action = JSON.parse('<?= $block->escapeHtml($block->getAction()); ?>');
```

File: view/frontend/templates/checkout/success.phtml [hunk 4]
```diff
@@ -25,7 +25,32 @@
...
+            var checkoutComponent = new AdyenCheckout({
+                locale: '<?php echo $block->getLocale(); ?>',
+                environment: '<?php echo $block->getEnvironment(); ?>',
+                originKey: '<?php echo $block->getOriginKey(); ?>'
```
Comment (cyattilakiss) [inline view/frontend/templates/checkout/success.phtml:40]:
```suggestion
                locale: '<?= $block->escapeHtml($block->getLocale()); ?>',
                environment: '<?= $block->escapeHtml($block->getEnvironment()); ?>',
                originKey: '<?= $block->escapeHtml($block->getOriginKey()); ?>'
```

File: view/frontend/templates/checkout/success.phtml [hunk 5]
```diff
@@ -25,7 +25,32 @@
...
+            });
+            var actionNode = document.getElementById('ActionContainer');
+            checkoutComponent.createFromAction(action).mount(actionNode);
```
Comment (cyattilakiss) [inline view/frontend/templates/checkout/success.phtml:43]: Remove the extra actionNode, checkout component can handle it
Put the createFromAction into a try catch

```suggestion
            try {
                checkoutComponent.createFromAction(action).mount('#ActionContainer');
            } catch(err) {
                // Action component cannot be created
            }
```

File: view/frontend/templates/checkout/success.phtml [hunk 6]
```diff
@@ -135,4 +162,5 @@
         <?php endif; ?>
         </tbody>
     </table>
-<?php endif; ?>
+    <?php endif; ?>
+<?php endif; ?>
```
Comment (cyattilakiss) [inline view/frontend/templates/checkout/success.phtml:166]:
```suggestion
<?php endif; ?>

```

File: view/frontend/templates/checkout/success.phtml (hunk 2 above)
Comment (rikterbeek) [reply view/frontend/templates/checkout/success.phtml:31]: No as it needs the php tags for the vraiables.

File: view/frontend/templates/checkout/success.phtml (hunk 1 above)
Comment (rikterbeek) [reply view/frontend/templates/checkout/success.phtml:56]: @acampos1916 this is not correct it need to be in the if statement

File: view/frontend/templates/checkout/success.phtml [hunk 7]
```diff
@@ -25,7 +25,28 @@
  * @var \Adyen\Payment\Block\Checkout\Success $block
  */
 ?>
-<?php if ($block->isBoletoPayment()): ?>
+<?php if ($block->renderAction()): ?>
+    <script type="text/javascript">
+        require([
+            'jquery',
+            'adyenCheckout3101'  // the alias for "mage/accordion"
```
Comment (acampos1916) [inline view/frontend/templates/checkout/success.phtml:32]:
```suggestion
            'adyenCheckout3101'
```

File: view/frontend/templates/checkout/success.phtml [hunk 8]
```diff
@@ -25,7 +25,28 @@
...
  */
 ?>
-<?php if ($block->isBoletoPayment()): ?>
+<?php if ($block->renderAction()): ?>
+    <script type="text/javascript">
+        require([
+            'jquery',
+            'adyenCheckout3101'
+        ], function ($, AdyenCheckout) {
+            var action = JSON.parse('<?= $block->getAction(); ?>');
```
Comment (cyattilakiss) [inline view/frontend/templates/checkout/success.phtml:34]: I don't know if we can escape it or not so to be sure that we don't change the structure of the action I would just add this:

```suggestion
            var action = JSON.parse('<?= /* @noEscape */ $block->getAction(); ?>');
```

File: Block/Checkout/Success.php [hunk 9]
```diff
@@ -151,14 +161,55 @@ public function getMultibancoData()
...
     }
 
+    /**
+     * If PresentToShopper resultCode and action has provided render this with the checkout component on the success page
+     * @return bool
+     */
+    public function renderAction()
+    {
+        if (
+            !empty($this->getOrder()->getPayment()->getAdditionalInformation('resultCode')) &&
```
Comment (msilvagarcia) [inline Block/Checkout/Success.php:171]: Is this `!empty` check even needed if you're comparing to `PresentToShopper` later?

File: view/frontend/templates/checkout/success.phtml (hunk 8 above)
Comment (msilvagarcia) [reply view/frontend/templates/checkout/success.phtml:34]: Would be nice to add some checks on the backend as well, then.

Comment (rikterbeek) [reply view/frontend/templates/checkout/success.phtml:34]: We can add this tag. @msilvagarcia not sure what you mean by add checks in the backend. As this comes from the API we just need to pass it to the component. The component will use it there will be no server call afterwards.

File: Block/Checkout/Success.php (hunk 9 above)
Comment (rikterbeek) [reply Block/Checkout/Success.php:171]: there is a potential that the resultCode does not exists, so that is why I added this check to prevent warnings.

**Pull request merged**